lib/adapters/http.js

```diff
@@ -524,6 +525,13 @@ export default isHttpAdapterSupported && function httpAdapter(config) {
             streams.push(zlib.createBrotliDecompress(brotliOptions));
             delete res.headers['content-encoding'];
           }
+          break;
+        case 'zstd':
+          if (isZstdSupported) {
+            streams.push(zlib.createZstdDecompress());
+            delete res.headers['content-encoding'];
```

Comment:
You should add the equivalent options like brotli does.

```js
const zstdOptions = {
  flush: zlib.constants.ZSTD_e_flush,
  finishFlush: zlib.constants.ZSTD_e_flush
}
```

Reply:
Sorry for delayed response, I added the relevant part to the commit [e2ba395](https://github.com/axios/axios/pull/6792/commits/e2ba395435f3f7f5260afffc8a3aad2d96159d06).